### PR TITLE
Load SnakeYaml in a separate DependencyClassLoader.

### PR DESCRIPTION
**Pull request by shroman**
moving Yaml conversion to a separate DependencyClassLoader.
Related PRs: #1201 (Netty Buffer) and #1204 (Airlift Slice).

Used in: org.embulk.config.ConfigLoader and org/embulk/EmbulkRunner

### CODE REVIEW

**Comment by shroman** [comment]
@dmikurube 
1. I completely forgot about `embulk-deps-config`. No reason, I will rename.
2. For conflicts, no worries, I am aware. After it's reviewed, I can always merge with the master branch.
For the order, can we have issues assigned to milestones, so that we can refer to them anytime?

File: embulk-core/src/main/java/org/embulk/deps/config/YamlProcessor.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
+package org.embulk.deps.config;
+
+import java.io.InputStream;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import org.embulk.deps.DependencyCategory;
+import org.embulk.deps.EmbulkDependencyClassLoaders;
+
+public abstract class YamlProcessor {
```
Comment (dmikurube) [inline embulk-core/src/main/java/org/embulk/deps/config/YamlProcessor.java]: I like the name `YamlProcessor`... SnakeYaml's `Yaml` was so cryptic! ;)

**Comment by dmikurube** [comment]
Ah, you already did it. :)

**Comment by shroman** [comment]
yeah, and something failed :)

**Comment by shroman** [comment]
missed it a couple of times ... :)

**Comment by dmikurube** [comment]
Merged!

**Pull request merged**